### ISSUE TITLE
CloudFormation: Delete stack containing IAM role with inline policy fails

### ISSUE THREAD

**Issue by mcintyre94** [localstack/localstack#2790]
# Type of request: This is a ...

[x] bug report
[ ] feature request

# Detailed description

After deploying a CloudFormation stack that contains an IAM role with an inline policy, I am unable to delete the stack using `cloudformation delete-stack`

## Expected behavior

Same as AWS proper: Stack and all its resources are deleted successfully

## Actual behavior

The stack is not deleted, nor are its resources.

```
$ awslocal cloudformation delete-stack --stack-name stack1

Unable to parse response (not well-formed (invalid token): line 1, column 0), invalid XML received:
b'{}'
```

From localstack logs, it seems to be trying to delete the role without first deleting the policy:

```
2020-07-30T09:53:47:DEBUG:localstack.utils.cloudformation.template_deployer: Running action "delete" for resource type "IAM::Role" id "ExecutionRole"
2020-07-30T09:53:47:DEBUG:localstack.utils.cloudformation.template_deployer: Request for resource type "IAM::Role" in region eu-west-1: delete_role {'RoleName': 'stack1-ExecutionRole-M0XZJRIXLJPE'}
2020-07-30T09:53:47:WARNING:localstack.utils.cloudformation.template_deployer: Error calling <bound method ClientCreator._create_api_method.<locals>._api_call of <botocore.client.IAM object at 0x7f223b88f8e0>> with params: {'RoleName': 'stack1-ExecutionRole-M0XZJRIXLJPE'} for resource: {'LogicalResourceId': 'ExecutionRole', 'PhysicalResourceId': 'stack1-ExecutionRole-M0XZJRIXLJPE', 'ResourceType': 'AWS::IAM::Role', 'LastUpdatedTimestamp': '2011-06-21 20:15:58+00:00', 'ResourceStatus': 'CREATE_COMPLETE', 'Properties': {'LogicalResourceId': 'ExecutionRole', 'PhysicalResourceId': 'stack1-ExecutionRole-M0XZJRIXLJPE', 'ResourceType': 'AWS::IAM::Role', 'LastUpdatedTimestamp': '2011-06-21 20:15:58+00:00', 'ResourceStatus': 'CREATE_COMPLETE'}}
2020-07-30T09:53:47:WARNING:localstack.utils.server.http2_server: Error in proxy handler for request POST http://localhost:4566/: An error occurred (DeleteConflict) when calling the DeleteRole operation: Cannot delete entity, must delete policies first. Traceback (most recent call last):
  File "/opt/code/localstack/localstack/utils/server/http2_server.py", line 53, in index
    result = await run_sync(handler, request, data)
  File "/opt/code/localstack/localstack/utils/async_utils.py", line 46, in run_sync
    return await loop.run_in_executor(THREAD_POOL, copy_context().run, func, *args)
  File "/usr/lib/python3.8/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/opt/code/localstack/localstack/services/generic_proxy.py", line 550, in handler
    response = modify_and_forward(method=method, path=path_with_params, data_bytes=data, headers=headers,
  File "/opt/code/localstack/localstack/services/generic_proxy.py", line 331, in modify_and_forward
    listener_result = listener.forward_request(method=method,
  File "/opt/code/localstack/localstack/services/cloudformation/cloudformation_listener.py", line 198, in forward_request
    template_deployer.delete_stack(stack_name, stack_resources)
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 1228, in delete_stack
    delete_resource(resource_id, resources, stack_name)
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 1078, in delete_resource
    return execute_resource_action(resource_id, resources, stack_name, ACTION_DELETE)
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 1100, in execute_resource_action
    result = configure_resource_via_sdk(resource_id, resources, resource_type, func, stack_name)
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 1177, in configure_resource_via_sdk
    raise e
  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 1174, in configure_resource_via_sdk
    result = function(**params)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/client.py", line 316, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/opt/code/localstack/.venv/lib/python3.8/site-packages/botocore/client.py", line 635, in _make_api_call
    raise error_class(parsed_response, operation_name)
botocore.errorfactory.DeleteConflictException: An error occurred (DeleteConflict) when calling the DeleteRole operation: Cannot delete entity, must delete policies first.
```

# Steps to reproduce

## Command used to start LocalStack

`DEFAULT_REGION=eu-west-1 DEBUG=1 localstack start`

## Client code (AWS SDK code snippet, or sequence of "awslocal" commands)

Template.yaml is a CloudFormation template containing one IAM role with one nested inline policy:

```yaml
AWSTemplateFormatVersion: '2010-09-09'

Resources:
  # IAM role for running the step function
  ExecutionRole:
    Type: "AWS::IAM::Role"
    Properties:
      AssumeRolePolicyDocument:
        Version: "2012-10-17"
        Statement:
        - Effect: "Allow"
          Principal:
            Service: !Sub states.${AWS::Region}.amazonaws.com
          Action: "sts:AssumeRole"
      Policies:
      - PolicyName: StatesExecutionPolicy
        PolicyDocument:
          Version: "2012-10-17"
          Statement:
          - Effect: Allow
            Action: "lambda:InvokeFunction"
            Resource: "*"
```

I deploy, and then try to delete this stack:

```
$ awslocal cloudformation deploy --stack-name stack1 --template-file template.yaml --capabilities CAPABILITY_IAM

Waiting for changeset to be created..
Waiting for stack create/update to complete
Successfully created/updated stack - stack1

$ awslocal cloudformation delete-stack --stack-name stack1

Unable to parse response (not well-formed (invalid token): line 1, column 0), invalid XML received:
b'{}'
```

**Comment by whummer** [comment]
Thanks for the detailed report @mcintyre94 . This should be fixed in #2804 - can you please do a `docker pull localstack/localstack` and give it another try with the `latest` version of the Docker image? Please report here if the problem persists. Thanks!

**Comment by mcintyre94** [comment]
@whummer Sorry I missed this one - fix looks good 👍
